plugins/system_upgrade.py

```diff
@@ -472,7 +472,10 @@ def run_upgrade(self, extcmds):
                 rpms.append(os.path.join(self.state.datadir, f))
         self.base.add_remote_rpms(rpms)
         # set up the upgrade transaction
-        self.base.upgrade_all()
+        if self.opts.distro_sync:
+            self.base.distro_sync()
```

Comment:
distro_sync is mysteriously working only when rootpath is not passed as parameter to Sack.